src/catalog/data/api.js

```diff
@@ -0,0 +1,13 @@
+import { ensureConfig, getConfig } from '@edx/frontend-platform';
+import { getAuthenticatedHttpClient } from '@edx/frontend-platform/auth';
+
+ensureConfig(['LMS_BASE_URL'], 'course API service');
+
+export const getCourses = async () => {
+  const client = getAuthenticatedHttpClient();
+  const baseUrl = getConfig().LMS_BASE_URL;
+  const response = await client.get(`${baseUrl}/api/courses/v1/courses/`);
+  return response.data.results;
```

Comment:
nit: I think this should just return  `response.data` since it also contains pagination information which may be of value. 

This is just a demo app so it's probably not a big deal though.